xds/src/main/java/io/grpc/xds/LoadStatsManager2.java

```diff
@@ -387,9 +387,13 @@ void release() {
     private ClusterLocalityStatsSnapshot snapshot() {
       long duration = stopwatch.elapsed(TimeUnit.NANOSECONDS);
       stopwatch.reset().start();
+      ImmutableMap<String, BackendLoadMetricStats> loadMetricStatsMapCopy;
+      synchronized (this) {
+        loadMetricStatsMapCopy = ImmutableMap.copyOf(loadMetricStatsMap);
+        loadMetricStatsMap.clear();
```

Comment:
Generally `clear()` is a bad idea for where we sit in the stack, as it means collections grow but never shrink which can be a security or reliability issue. Some places we'll actually use LinkedList instead of ArrayList because of that. I think we want to replace it with a new one (`loadMetricStatsMap = new HashMap<>()`). Yes, that means the field can't be `final`.

The copying by ImmutableMap isn't much concern since it happens rarely, but I think we just get nothing from using ImmutableMap above just using unmodifiableMap. Especially with mutable values. I think we're probably better of using unmodifiableMap(), but I could live with the ImmutableMap usage. Meh.

Reply:
Done.
1. Replaced `loadMetricStatsMap` with a new one using `new HashMap<>()` instead of `clear()`.
2. Used `Collections.unmodifiableMap()` for copying instead of `ImmutableMap.copyOf()`

Extra contexts & investigations:
For point 2, using `Collections.unmodifiableMap()` is effectively immutable and safe in our case because **_nobody holds a reference to the original backing collection_** later (as this was done with point 1 using `loadMetricStatsMap = new HashMap<>()`) [1, 2, 3, 4]


References
[1] https://docs.oracle.com/javase/tutorial/collections/implementations/wrapper.html#Unmodifiable_Wrappers
[2]https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Collections.html#unmodifiableMap(java.util.Map)
[3] https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Collection.html
[4] https://github.com/google/guava/wiki/ImmutableCollectionsExplained